### PR TITLE
Part of #12098: Use ScreenRect on gfx_filter_rect

### PR DESCRIPTION
**Pull request by annoyingdroid**
Changed another reference to the old `gfx_filter_rect` function. I also found that the remaining references to the old function are located in the `gfx_fill_rect_inset`. 

When I attempt to change them, I receive an error stating `no instance of overloaded function gfx_filter_rect matches the argument list`. My guess is that once issue #12097 is completed, we'll be able to get those taken care of.

### CODE REVIEW

**Comment by annoyingdroid** [comment]
I didn't realize Visual Studio had the find references feature (I had simply been using the search feature and typing in the function name). If that's the case. I don't mind closing this pull request and simply resubmitting when I have more code changes to merge with the project. 

As for the commit history, I'm not entirely sure what had happened? I had carried out a rebase prior to the push, though I can go ahead and get that cleaned up before the next pull request I make.

**Comment by tupaschoal** [comment]
> I didn't realize Visual Studio had the find references feature (I had simply been using the search feature and typing in the function name). If that's the case. I don't mind closing this pull request and simply resubmitting when I have more code changes to merge with the project.

Please don't close one PR and open another one which is almost the same or similar. Just do the work on this one. It's ok if it takes some days :). Another thing I can do is mark this as a "draft PR" and when it's ready you click on ready for review, so we know you're done.

> As for the commit history, I'm not entirely sure what had happened? I had carried out a rebase prior to the push, though I can go ahead and get that cleaned up before the next pull request I make.

Weird, you can always squash them together too.

**Comment by annoyingdroid** [comment]
Feel free to mark this as a draft PR in that case! It will take me a little time to make additional code changes.

**Comment by tupaschoal** [comment]
Done

**Comment by tupaschoal** [comment]
@annoyingdroid do you intend on resuming this?

**Comment by annoyingdroid** [comment]
That was my intention, though I've gotten really busy with other work as of late. I don't mind if you want to close this request or just have someone else work on it.

**Comment by tupaschoal** [comment]
It's alright, if you intend to I'll leave it open

**Comment by annoyingdroid** [comment]
Still not entirely sure of how to get rid of those extra commits that keep showing up. I'm rebasing the same way as usual and those keep showing up. Fixing the code formatting again too.

**Comment by tupaschoal** [comment]
Which extra commits? Do remember that you first have to switch to your develop branch, sync it with the main repo, then go back to your branch to rebase
https://github.com/OpenRCT2/OpenRCT2/wiki/Rebase-and-Sync-fork-with-OpenRCT2

**Comment by annoyingdroid** [comment]
Oh! I thought that those commits from 29 days ago showed back up on this latest push. Looks like it's all good now.

File: src/openrct2-ui/interface/Widget.cpp [hunk 1]
```diff
@@ -532,7 +532,7 @@ static void widget_caption_draw(rct_drawpixelinfo* dpi, rct_window* w, rct_widge
         gfx_fill_rect(
             dpi, { { topLeft + ScreenCoordsXY{ 1, 1 } }, { bottomRight - ScreenCoordsXY{ 1, 1 } } }, ColourMapA[colour].dark);
     else
-        gfx_filter_rect(dpi, topLeft.x + 1, topLeft.y + 1, bottomRight.x - 1, bottomRight.y - 1, PALETTE_DARKEN_3);
+        gfx_filter_rect(dpi, { { topLeft.x + 1, topLeft.y + 1 }, { bottomRight.x - 1, bottomRight.y - 1 } }, PALETTE_DARKEN_3);
```
Comment (tupaschoal) [inline src/openrct2-ui/interface/Widget.cpp:535]:
```suggestion
        gfx_filter_rect(dpi, { topLeft + ScreenCoordsXY{ 1, 1 }, { bottomRigh - ScreenCoordsXY{ 1, 1 } }, PALETTE_DARKEN_3);
```

File: src/openrct2-ui/interface/Window.cpp [hunk 2]
```diff
@@ -587,7 +587,8 @@ void window_draw_widgets(rct_window* w, rct_drawpixelinfo* dpi)
 
     if ((w->flags & WF_TRANSPARENT) && !(w->flags & WF_NO_BACKGROUND))
         gfx_filter_rect(
-            dpi, w->windowPos.x, w->windowPos.y, w->windowPos.x + w->width - 1, w->windowPos.y + w->height - 1, PALETTE_51);
+            dpi, { { w->windowPos.x, w->windowPos.y }, { w->windowPos.x + w->width - 1, w->windowPos.y + w->height - 1 } },
```
Comment (tupaschoal) [inline src/openrct2-ui/interface/Window.cpp:590]:
```suggestion
            dpi, { w->windowPos, w->windowPos + ScreenCoordsXY{ w->width - 1, w->height - 1 } },
```

File: src/openrct2-ui/interface/Window.cpp [hunk 3]
```diff
@@ -587,7 +587,7 @@ void window_draw_widgets(rct_window* w, rct_drawpixelinfo* dpi)
 
     if ((w->flags & WF_TRANSPARENT) && !(w->flags & WF_NO_BACKGROUND))
         gfx_filter_rect(
-            dpi, w->windowPos.x, w->windowPos.y, w->windowPos.x + w->width - 1, w->windowPos.y + w->height - 1, PALETTE_51);
+            dpi, { w->windowPos, w->windowPos + ScreenCoordsXY{ w->width - 1, w->windowPos.y + w->height - 1 } }, PALETTE_51);
```
Comment (tupaschoal) [inline src/openrct2-ui/interface/Window.cpp:590]: There's an extra window pos on y arg here

Comment (AaronVanGeffen) [reply src/openrct2-ui/interface/Window.cpp:590]: Nice! Looks like that's the issue indeed.

Comment (annoyingdroid) [reply src/openrct2-ui/interface/Window.cpp:590]: Issue should be fixed. Deleted the extra windowPos and tested locally. I'll make another pull request.

Comment (AaronVanGeffen) [reply src/openrct2-ui/interface/Window.cpp:590]: I already filed #12554, as this is a game breaker.

**Pull request merged**